### PR TITLE
Add thanos resources

### PR DESCRIPTION
**Pull request by ankushbbbr**
- Add Thanos spec in prometheus CR
- Add Thanos Querier deployment and service
- Add config to enable / disable thanos

### CODE REVIEW

File: pkg/core/core.go [hunk 1]
```diff
@@ -362,6 +363,9 @@ func GetPrometheusObject(plantDCore *windtunnelv1alpha1.PlantDCore) *monitoringv
 		resourceRequirements.Limits[corev1.ResourceMemory] = resource.MustParse(prometheusMemoryLimit)
 	}
 
+	thanosBaseImage := "quay.io/thanos/thanos"
+	thanosVersion := "v0.34.0"
```
Comment (baljit92) [inline pkg/core/core.go]: Can we move these to config.go?

File: pkg/core/core.go [hunk 2]
```diff
@@ -483,3 +503,98 @@ func GetRedisService(plantDCore *windtunnelv1alpha1.PlantDCore) *corev1.Service
...
+			Namespace: plantDCore.Namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeClusterIP,
+			Ports: []corev1.ServicePort{
+				{
+					Name:       "http",
+					Protocol:   corev1.ProtocolTCP,
+					Port:       9090,
+					TargetPort: intstr.FromInt(9090),
```
Comment (baljit92) [inline pkg/core/core.go]: Move them to _config.go_

File: pkg/core/core.go [hunk 3]
```diff
@@ -483,3 +503,98 @@ func GetRedisService(plantDCore *windtunnelv1alpha1.PlantDCore) *corev1.Service
...
+				},
+			},
+			Selector: map[string]string{
+				"app": "thanos-querier",
```
Comment (baljit92) [inline pkg/core/core.go]: Move them to _config.go_

File: pkg/core/core.go [hunk 4]
```diff
@@ -483,3 +503,98 @@ func GetRedisService(plantDCore *windtunnelv1alpha1.PlantDCore) *corev1.Service
...
+			Namespace: plantDCore.Namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeClusterIP,
+			Ports: []corev1.ServicePort{
+				{
+					Name:       "grpc",
+					Port:       10901,
+					TargetPort: intstr.FromInt(10901),
+					Protocol:   corev1.ProtocolTCP,
```
Comment (baljit92) [inline pkg/core/core.go]: Move them to config.go

File: pkg/core/core.go [hunk 5]
```diff
@@ -483,3 +503,98 @@ func GetRedisService(plantDCore *windtunnelv1alpha1.PlantDCore) *corev1.Service
...
+				},
+			},
+			Selector: map[string]string{
+				"prometheus": "prometheus",
```
Comment (baljit92) [inline pkg/core/core.go]: Move to config.go

File: pkg/core/core.go [hunk 6]
```diff
@@ -483,3 +503,98 @@ func GetRedisService(plantDCore *windtunnelv1alpha1.PlantDCore) *corev1.Service
...
+							},
+							Ports: []corev1.ContainerPort{
+								{
+									Name:          "http",
+									ContainerPort: 9090,
+								},
+								{
+									Name:          "grpc",
+									ContainerPort: 10901,
+								},
```
Comment (baljit92) [inline pkg/core/core.go]: Move them to config.go

File: pkg/core/core.go [hunk 7]
```diff
@@ -378,11 +382,28 @@ func GetPrometheusObject(plantDCore *windtunnelv1alpha1.PlantDCore) *monitoringv
...
-
+	if plantDCore.Spec.ThanosEnabled {
+		prometheus.Spec.Thanos = &monitoringv1.ThanosSpec{
+			BaseImage: &thanosBaseImage,
+			Version:   &thanosVersion,
+			ObjectStorageConfig: &corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: "thanos-objstore-config",
+				},
+				Key: "thanos.yaml",
```
Comment (baljit92) [inline pkg/core/core.go]: Can we move this to config.go?

**Review by baljit92 — CHANGES_REQUESTED** [review]

File: pkg/core/core.go [hunk 8]
```diff
@@ -418,7 +439,6 @@ func GetRedisDeployment(plantDCore *windtunnelv1alpha1.PlantDCore) *appsv1.Deplo
 	if image == "" {
 		image = redisImage
 	}
-
```
Comment (tomzhu1024) [inline pkg/core/core.go]: Is this intentional, or a mistake?

**Comment by ankushbbbr** [comment]
Moved configurable parameters to config.yaml

**Review by tomzhu1024 — APPROVED** [review]

**Pull request merged**